fuel/streams.py

```diff
@@ -135,3 +138,42 @@ def get_epoch_iterator(self, **kwargs):
         else:
             self._fresh_state = False
         return super(DataStream, self).get_epoch_iterator(**kwargs)
+
+
+class ServerDataStream(AbstractDataStream):
```

Comment:
Yeah, there's a bunch of things that strictly speaking need to be communicated: Getting a new iterator, `num_examples` and `num_batches`, resetting the state, etc.

If you can find a particularly clean way to do it, let me know, although I think it won't be too bad to do it the naive way. Right now the server responds to every message with a batch, but it could easily check the message and send different data instead.